anxcloud/data_source_core_location.go

```diff
@@ -0,0 +1,78 @@
+package anxcloud
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+)
+
+func dataSourceCoreLocation() *schema.Resource {
+	return &schema.Resource{
+		Description: "Retrieves Location identified by it's `code`.",
+		ReadContext: dataSourceCoreLocationRead,
+		Schema:      schemaLocation(),
+	}
+}
+
+func dataSourceCoreLocationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	a := m.(providerContext).api
+
+	code, exists := d.GetOk("code")
+	if !exists {
+		return diag.Errorf("location data-source requires code argument")
+	}
+
+	searchLocation := &corev1.Location{Code: code.(string)}
+	channel := make(types.ObjectChannel)
+	if err := a.List(ctx, searchLocation, api.ObjectChannel(&channel)); err != nil {
```

Comment:
when `List`ing with object channels, you always have to either read until the channel is closed or use a context you cancel when you found your object - without that, you'll leak a goroutine.

https://pkg.go.dev/go.anx.io/go-anxcloud@v0.4.2/pkg/api#ObjectChannel